File: TpWeb-main/angular/scholarIA/src/app/services/like.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface LikeCreationRequest {
  researchPaperId: string;
  [key: string]: any;
}

export interface LikeResponse {
  id: string;
  paperId: string;
  userId?: string;
  createdAt?: string;
  [key: string]: any;
}

@Injectable({
  providedIn: 'root'
})
export class LikeService {
  private apiUrl = 'http://localhost:8080/api/v1/like';

  constructor(private http: HttpClient) { }

  /**
   * Add a like to a paper
   */
  addLike(request: LikeCreationRequest): Observable<void> {
    return this.http.post<void>(`${this.apiUrl}/add`, request);
  }

  /**
   * Get like by ID
   */
  getLikeById(likeId: string): Observable<LikeResponse> {
    return this.http.get<LikeResponse>(`${this.apiUrl}/${likeId}`);
  }

  /**
   * Get all likes by user ID
   */
  getAllLikesByUserId(userId: string): Observable<LikeResponse[]> {
    return this.http.get<LikeResponse[]>(`${this.apiUrl}/${userId}`);
  }

  /**
   * Get all likes by paper ID
   */
  getAllLikesByPaperId(paperId: string): Observable<LikeResponse[]> {
    return this.http.get<LikeResponse[]>(`${this.apiUrl}/${paperId}`);
  }
}

File: TpWeb-main/back/src/main/java/com/research/paper/service/feedback/LikeService.java
package com.research.paper.service.feedback;

import com.research.paper.dto.request.feedback.likes.LikeCreationRequest;
import com.research.paper.dto.response.feedback.LikeResponse;

import java.util.List;

public interface LikeService {
    void addLike(LikeCreationRequest request, String userId);
    void deleteLike(String LikeId);
    LikeResponse getLikeById(String LikeId);
    List<LikeResponse> getAllLikesByPaperId(String paperId);
    List<LikeResponse> getAllLikesByUserId(String userId);
}
